core/storage/base_model/gae_models.py

```diff
@@ -542,17 +540,34 @@ def get_multi_versions(cls, entity_id, version_numbers):
         Returns:
             list(VersionedModel). Model instances representing the given
                 versions.
+
+        Raises:
+            ValueError. Given entity_id is invalid.
+            ValueError. Requested version number cannot be higher than the
+                current version number.
+            ValueError. Version number is invalid.
         """
         instances = []
-        try:
-            cls.get(entity_id)
-        except Exception as  e:
-            logging.error(e)
+
+        entity = cls.get(entity_id, strict=False)
+        if not entity:
+            raise ValueError('Given entity_id %s is invalid.' % (entity_id))
+        current_version = entity.version
+        if version_numbers[-1] > current_version:
```

Comment:
Do not assume that the version numbers are sorted.

Reply:
Done. Using max() instead now.